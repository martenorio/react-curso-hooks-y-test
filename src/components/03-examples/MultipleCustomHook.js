import React from 'react'
import { useFetch } from '../../hooks/useFetch';
import { useCounter } from '../../hooks/useCounter';
import '../02-useEffect/effects.css';

export const MultipleCustomHook = () => {
    const { counter, increment } = useCounter();
    const { loading, data } = useFetch(`https://www.breakingbadapi.com/api/quotes/${counter}`);
    const { author, quote } = !!data && data[0];
    return (
        <div>
            <h1>Breaking Quote</h1>
            <hr />
            {
                loading ?
                (
                    <div className='alert alert-info'>loading...</div>
                ) :
                (
                    <div>
                        <blockquote className='blockquote text-end'>
                            <p className='mb-0'>{ quote }</p>
                            <footer className='blockquote-footer'>{ author }</footer>
                        </blockquote>
                        <button onClick={increment} className='btn btn-primary'>Siguiente quote</button>
                    </div>
                )
            }
        </div>
    )
}
