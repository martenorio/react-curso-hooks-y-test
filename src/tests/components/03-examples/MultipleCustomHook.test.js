import { shallow } from "enzyme";
import { MultipleCustomHook } from '../../../components/03-examples/MultipleCustomHook';
import { useCounter } from "../../../hooks/useCounter";
import { useFetch } from "../../../hooks/useFetch";
jest.mock("../../../hooks/useFetch");
jest.mock("../../../hooks/useCounter");


describe('MultipleCustomHook tests', () => {
    beforeEach( () =>  {
        useCounter.mockReturnValue({
            counter:10,
            increment: () => {}
        });
    })
    test('should show snapshot correctly', () => {
        useFetch.mockReturnValue({
            data:null,
            loading:true,
            error:null
        });
        const wrapper = shallow(<MultipleCustomHook />);
        expect(wrapper).toMatchSnapshot();
    });
    test('should show loaded information', () => {
        useFetch.mockReturnValue({
            data:[{author:"Miguel",quote:"Holi"}],
            loading:false,
            error:null
        });
        const wrapper = shallow(<MultipleCustomHook />);
        expect( wrapper.find('.alert').exists() ).toBe(false);
        expect( wrapper.find('.mb-0').text().trim() ).toBe("Holi");
        expect( wrapper.find('footer').text().trim() ).toBe("Miguel");
        expect(wrapper).toMatchSnapshot();
    })
})