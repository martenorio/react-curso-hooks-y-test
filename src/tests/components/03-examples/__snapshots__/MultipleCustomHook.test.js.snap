// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MultipleCustomHook tests should show loaded information 1`] = `
<div>
  <h1>
    Breaking Quote
  </h1>
  <hr />
  <div>
    <blockquote
      className="blockquote text-end"
    >
      <p
        className="mb-0"
      >
        Holi
      </p>
      <footer
        className="blockquote-footer"
      >
        Miguel
      </footer>
    </blockquote>
    <button
      className="btn btn-primary"
      onClick={[Function]}
    >
      Siguiente quote
    </button>
  </div>
</div>
`;

exports[`MultipleCustomHook tests should show snapshot correctly 1`] = `
<div>
  <h1>
    Breaking Quote
  </h1>
  <hr />
  <div
    className="alert alert-info"
  >
    loading...
  </div>
</div>
`;

exports[`MultipleCustomHook tests should show snapshot successfully 1`] = `
<div>
  <h1>
    Breaking Quote
  </h1>
  <hr />
  <div
    className="alert alert-info"
  >
    loading...
  </div>
</div>
`;
